File: 999. Available Captures for Rook/numRookCaptures.py
class numRookCaptures:
    def solution(self, board):
        count = 0
        row = 0
        col = 0
        for i in range(0,8):
            rowStr = ''.join(board[i])
            if rowStr.find('R') > 0:
                row = i
                col = rowStr.find('R')
                break
        rowStr = rowStr.replace('.', '')
        if 'pR' in rowStr:
            count = count + 1
        if 'Rp' in rowStr:
            count = count + 1
        colStr = ''.join(c[col] for c in board).replace('.', '')
        if 'pR' in colStr:
            count = count + 1
        if 'Rp' in colStr:
            count = count + 1
        return count


arr = [[".",".",".",".",".",".",".","."],[".",".",".","p",".",".",".","."],[".",".",".","R",".",".",".","p"],[".",".",".",".",".",".",".","."],[".",".",".",".",".",".",".","."],[".",".",".","p",".",".",".","."],[".",".",".",".",".",".",".","."],[".",".",".",".",".",".",".","."]]
obj = numRookCaptures()
res = obj.solution(arr)
print(res)

File: 999. Available Captures for Rook/numRookCaptures.java
class numRookCaptures {
    public int solution(char[][] board) {
    	int count = 0;
    	int[] dx = {-1, 1, 0, 0};
    	int[] dy = {0, 0, -1, 1};

    	for (int i = 0; i < 8; i++) {
    		for (int j = 0; j < 8; j++) {
    			if (board[i][j] == 'R') {
    				for (int k = 0; k < 4; k++) {
    					int x = i;
    					int y = j;
    					while (true) {
    						x = x + dx[k];
    						y = y + dy[k];
    						if ( x < 0 || x >= 8 || y < 0 || y >= 8 || board[x][y] == 'B') {
    							break;
    						}
    						if (board[x][y] == 'p') {
    							count = count + 1;
    							break;
    						}
    					}
    				}
    				return count;
    			}
    		}
    	}
    	return count;
    }

    public static void main(String[] argv) {
    	char[][] arr = {{'.','.','.','.','.','.','.','.'},{'.','.','.','p','.','.','.','.'},{'.','.','.','p','.','.','.','.'},{'p','p','.','R','.','p','B','.'},{'.','.','.','.','.','.','.','.'},{'.','.','.','B','.','.','.','.'},{'.','.','.','p','.','.','.','.'},{'.','.','.','.','.','.','.','.'}};
    	// char[][] arr = {{'.','.','.','.','.','.','.','.'},{'.','.','.','p','.','.','.','.'},{'.','.','.','R','.','.','.','p'},{'.','.','.','.','.','.','.','.'},{'.','.','.','.','.','.','.','.'},{'.','.','.','p','.','.','.','.'},{'.','.','.','.','.','.','.','.'},{'.','.','.','.','.','.','.','.'}};
    	numRookCaptures obj = new numRookCaptures();
    	int res = obj.solution(arr);
    	System.out.println(res);
    }
}